python/lsst/ip/diffim/getTemplate.py

```diff
@@ -107,19 +113,32 @@ def run(self, exposure, sensorRef, templateIdList=None):
                 bbox=patchSubBBox,
                 tract=tractInfo.getId(),
                 patch="%s,%s" % (patchInfo.getIndex()[0], patchInfo.getIndex()[1]),
+                numSubfilters=self.config.numSubfilters,
             )
             if patchSubBBox.isEmpty():
                 self.log.info("skip tract=%(tract)s, patch=%(patch)s; no overlapping pixels" % patchArgDict)
                 continue
-            if not sensorRef.datasetExists(**patchArgDict):
-                self.log.warn("%(datasetType)s, tract=%(tract)s, patch=%(patch)s does not exist"
-                              % patchArgDict)
-                continue
 
             nPatchesFound += 1
```

Comment:
nPatchesFound was incremented if the patch was found by the butler. This way if there was no data to make a template, it would raise.  I think you meant to put your block that reads or builds a patch above nPatchedFound += 0.